hack/extract.go

```diff
@@ -0,0 +1,253 @@
+	e := filepath.Walk(translationsFiles, func(path string, info os.FileInfo, err error) error {
+		if info.Mode().IsDir() {
+			return nil
+		}
+		fmt.Printf("Writing to %s\n", filepath.Base(path))
+		var currentTranslations map[string]interface{}
+		f, err := ioutil.ReadFile(path)
+		if err != nil {
+			return err
+		}
+		err = json.Unmarshal(f, &currentTranslations)
+		if err != nil {
+			return err
+		}
+
+		//fmt.Println(currentTranslations)
+
+		for k := range translations {
+			//fmt.Println(k)
+			if _, ok := currentTranslations[k]; !ok {
+				currentTranslations[k] = " "
+			}
+		}
+
+		c, err := json.MarshalIndent(currentTranslations, "", "\t")
+		if err != nil {
+			return err
+		}
+		err = ioutil.WriteFile(path, c, info.Mode())
+		return err
+	})
+
+	return e
+}
+
+func addFuncToList(f string) {
+	if _, ok := funcs[f]; !ok {
+		funcs[f] = struct{}{}
+		funcStack.Push(f)
+	}
+}
+
+func shouldCheckFile(path string) bool {
+	return strings.HasSuffix(path, ".go") && !strings.HasSuffix(path, "_test.go")
+}
+
+func inspectFile(filename string, f string) error {
+	fset := token.NewFileSet()
+	r, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	//fmt.Printf("    Parsing %s\n", filename)
+	file, err := parser.ParseFile(fset, "", r, parser.ParseComments)
+	if err != nil {
+		return err
+	}
+
+	ast.Inspect(file, func(x ast.Node) bool {
+		fd, ok := x.(*ast.FuncDecl)
+		if !ok {
+			/*gd, ok := x.(*ast.GenDecl)
+			if !ok {
+				return true
+			}
+			for _, spec := range gd.Specs {
+				if vs, ok := spec.(*ast.ValueSpec); ok {
+					for _, v := range vs.Values {
+						if ue, ok := v.(*ast.UnaryExpr); ok {
+							fmt.Printf("%s: %s\n", ue.X, reflect.TypeOf(ue.X))
+						}
+					}
+				}
+			}*/
+			return true
+		}
+
+		for _, stmt := range fd.Body.List {
+			checkStmt(stmt, fd.Name.String(), f)
+		}
+		return true
+	})
+
+	return nil
+}
+
+func checkStmt(stmt ast.Stmt, parentFunc string, f string) {
+	// If this line is an expression, see if it's a function call
+	if t, ok := stmt.(*ast.ExprStmt); ok {
+		checkCallExpression(t, parentFunc, f)
+	}
+
+	// If this line is the beginning of an if statement, then check of the body of the block
+	if b, ok := stmt.(*ast.IfStmt); ok {
+		checkIfStmt(b, parentFunc, f)
+	}
+
+	// Same for loops
+	if b, ok := stmt.(*ast.ForStmt); ok {
+		for _, s := range b.Body.List {
+			checkStmt(s, parentFunc, f)
+		}
+	}
+}
+
+func checkIfStmt(stmt *ast.IfStmt, parentFunc string, f string) {
+	for _, s := range stmt.Body.List {
+		checkStmt(s, parentFunc, f)
+	}
+	if stmt.Else != nil {
+		// A straight else
+		if block, ok := stmt.Else.(*ast.BlockStmt); ok {
+			for _, s := range block.List {
+				checkStmt(s, parentFunc, f)
+			}
+		}
+
+		// An else if
+		if elseif, ok := stmt.Else.(*ast.IfStmt); ok {
+			checkIfStmt(elseif, parentFunc, f)
+		}
+
+	}
+}
+
+func checkCallExpression(t *ast.ExprStmt, parentFunc string, f string) {
+	if s, ok := t.X.(*ast.CallExpr); ok {
```

Comment:
Try to keep the normal code path at a minimal indentation by terminating the exceptional condition early on. For instance:

```
if s, ok := t.X.(*ast.CallExpr); !ok {
  return
}
.. do fun things! ...
```

It takes a few more lines, but it'll be easier for new readers to follow the code path.